### PR TITLE
[SPARK-35734][SQL][FOLLOWUP] IntervalUtils.toDayTimeIntervalString should consider the case a day-time type is casted as another day-time type

### PR DESCRIPTION
**Pull request by sarutak**
### What changes were proposed in this pull request?

This PR fixes an issue that `IntervalUtils.toDayTimeIntervalString` doesn't consider the case that a day-time interval type is casted as another day-time interval type.
if data of `interval day to second` is casted as `interval hour to second`, the value of the day is multiplied by 24 and added to the value of hour. For example, `INTERVAL '1 2' DAY TO HOUR` will be `INTERVAL '26' HOUR` if it's casted.
If this behavior is intended, it should be stringified as `INTERVAL '26' HOUR` but currently, it will be `INTERVAL '2' HOUR`

### Why are the changes needed?

t's a bug if the behavior of cast is intended.

### Does this PR introduce _any_ user-facing change?

No, because this feature is not released yet.

### How was this patch tested?

Modified the tests added in SPARK-35734 (#32891)

### CODE REVIEW

**Comment by MaxGekk** [comment]
@sarutak Could you fix build by scala 2.13:
```
[error] /home/runner/work/spark/spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:1107:34: value format is not a member of StringBuilder
[error]         s"$prefix${formatBuilder.format(formatArgs.toSeq: _*)}$postfix"
[error]                                  ^
[error] one error found
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala [hunk 1]
```diff
@@ -1023,83 +1023,95 @@ object IntervalUtils {
...
-          startField match {
-            case DayTimeIntervalType.DAY => formatBuilder.append(s"$sign$days' ")
-            case DayTimeIntervalType.HOUR => formatBuilder.append(f"$hours%02d' ")
-            case DayTimeIntervalType.MINUTE => formatBuilder.append(f"$minutes%02d' ")
-            case DayTimeIntervalType.SECOND => formatBuilder.append(s"$leadSecZero$secStr' ")
-          }
-          formatBuilder.append(from).toString
+        val formatBuilder = new mutable.StringBuilder(sign)
+        val formatArgs = new mutable.ArrayBuffer[Long]()
+        if (startField == DayTimeIntervalType.SECOND) {
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:1072]: I didn't get why do you check `SECOND` in the if but the rest fields in `match .. case`. Can you handle `SECOND` in the same way as others?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala [hunk 2]
```diff
@@ -1023,83 +1023,95 @@ object IntervalUtils {
...
-            val fromPos = startField match {
-              case DayTimeIntervalType.DAY => 0
-              case DayTimeIntervalType.HOUR => 11
-              case DayTimeIntervalType.MINUTE => 14
-              case DayTimeIntervalType.SECOND => 17
+            val firstStr = startField match {
+              case DayTimeIntervalType.DAY => "-106751991"
+              case DayTimeIntervalType.HOUR => "-2562047788"
+              case DayTimeIntervalType.MINUTE => "-153722867280"
+              case DayTimeIntervalType.SECOND => "-9223372036854.775808"
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:1041]: Could you import field names DAY .. SECOND, and remove `DayTimeIntervalType`. It just makes reading harder.

Comment (sarutak) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala:1041]: Hmm, `IntervalUnit` has the same name of members and `IntervalUtils` imports them. So, importing `DAY`, `SECOND` or any other units and just refers them can be cofusable right?
Instead, how about importing `DayTimeIntervalType` as `DT` and refer the units like `DT.DAY`?

**Review by MaxGekk — APPROVED** [review]
+1, LGTM. Merging to master.
Thank you, @sarutak .

**Pull request closed without merging**